addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -301,37 +333,88 @@ private void handleMute(final Command command) {
         }
     }
 
-    private void scheduleStopApp() {
-        scheduler.schedule(() -> {
-            try {
-                Application app = chromecast.getRunningApp();
-                if (app.id.equals(MEDIA_PLAYER) && this.appSessionId != null
-                        && app.sessionId.equals(this.appSessionId)) {
-                    chromecast.stopApp();
-                    logger.debug("Media player app stopped");
-                }
-            } catch (final Exception e) {
-                logger.debug("Failed stopping media player app: {}", e.getMessage());
-            }
-        }, 0, TimeUnit.MILLISECONDS);
+    @Override
+    public void spontaneousEventReceived(final ChromeCastSpontaneousEvent event) {
+        switch (event.getType()) {
+            case MEDIA_STATUS:
+                final MediaStatus mediaStatus = event.getData(MediaStatus.class);
+                updateMediaStatus(mediaStatus);
+                break;
+            case STATUS:
+                final Status status = event.getData(Status.class);
+                updateChromecastStatus(status);
+                break;
+            case UNKNOWN:
+                logger.debug("Received an 'UNKNOWN' event (class={})", event.getType().getDataClass());
+                break;
+            default:
+                logger.debug("Unhandled event type: {}", event.getData());
+                break;
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, String description) {
+        if (status == ThingStatus.ONLINE) {
+            updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_ONLINE_STATUS),
+                    OnOffType.ON);
+        } else {
+            updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_ONLINE_STATUS),
+                    OnOffType.OFF);
+        }
+        super.updateStatus(status, statusDetail, description);
     }
 
-    private void handleCcStatus(final Status status) {
+    private void updateChromecastStatus(final Status status) {
         logger.debug("STATUS {}", status);
         if (status.applications == null) {
             this.appSessionId = null;
         }
-        handleCcVolume(status.volume);
+
+        updateStatus(ThingStatus.ONLINE);
+        updateAppStatus(status.getRunningApp());
+        updateVolumeStatus(status.volume);
+    }
+
+    private void updateAppStatus(final Application application) {
+        String name = "";
+        String id = "";
+        String statusText = "";
+        OnOffType idling = OnOffType.OFF;
+
+        if (application != null) {
```

Comment:
But if the application is null, is this useful? Should we just cancel updating the state?

idling = OFF => sounds like active to me?
If you want to update the states for the others UNDEFINED might be more appropriate?